decisions/2021/04/21-fe-ui-meeting.md

```diff
@@ -0,0 +1,44 @@
+When: April 21, 2021 @ 13:00 CET
+Where: Remote
+Who: Hendrik, Médi-Rémi, Edoardo, Ismay, Kai, Jan-Gerke, Austin
+
+# Agenda
+
+## Programmatic handling of state
+
+-   No documentation on the current way of handling callbacks
+
+-   Programmatic handling of open/close states
+
+    -   Prefer props over state
+
+-   Consider the cost/benefit of breaking compatibility.
+
+SUGGESTION: We could have different signatures for ui-elements and
+ui-composes.
+
+SUGGESTION: For next major version of UI switch event signature for
+callbacks to `(event, payload) => {}`.
+
+DECISION: To avoid breaking the callback signature for components when
+refactoring them into "style components", we can wrap the on-handlers in
+the components so they use the `(payload, event)` signature but still
+allow prop-spreading and forwardRef.
```

Comment:
Just to check: this means we keep the current component implementation in tact but add props spreading and forwardRef?

Reply:
Yes, however, changing existing components to forward refs is recommended to be a breaking change from the React docs: https://reactjs.org/docs/forwarding-refs.html#note-for-component-library-maintainers

>When you start using forwardRef in a component library, you should treat it as a breaking change and release a new major version of your library.

So it is something we would want to add to everything at once when we do it.